uk.co.compucorp.civicrm.hrcore/tests/phpunit/CRM/HRCore/CMSData/UserMailNotifierFactoryTest.php

```diff
@@ -11,17 +11,24 @@
 class CRM_HRCore_CMSData_CMSUserMailNotifierFactoryTest extends CRM_HRCore_Test_BaseHeadlessTest {
 
   public function testItReturnsAnInstanceOfTheExpectedClassWhenCMSIsSupported() {
-    $contactData = ['cmsId' => 1];
-    $mailNotifier = UserMailNotifierFactory::create('Drupal', $contactData);
+    //We need to manually set this value because civicrm sets it to UnitTests
+    //when running in testing environment.
+    CRM_Core_Config::singleton()->userFramework = 'Drupal';
+    $mailNotifier = UserMailNotifierFactory::create();
     $this->assertInstanceOf(UserMailNotifierInterface::class, $mailNotifier);
+    CRM_Core_Config::singleton()->userFramework = 'UnitTests';
```

Comment:
I think a better approach for this would be:

```php

$previousUserFramework = CRM_Core_Config::singleton()->userFramework = 'UnitTests';
CRM_Core_Config::singleton()->userFramework = 'Drupal';
// test steps
CRM_Core_Config::singleton()->userFramework = $previousUserFramework;
```

This way, we avoid problems if in the future they change `UnitTests` to something else